Training/Training/intlist.c

```diff
@@ -0,0 +1,285 @@
+int RemoveAt (struct LinkedList* list, int ind) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* temp = list->head;
+	struct Node* prev = NULL;
+
+	// Base case if linked list is empty
+	if (temp == NULL)
+		return E_EMPTY_LIST;
+
+	// Case 1: Head is to be deleted
+	if (ind == 0) {
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+
+	// Case 2: Node to be deleted is in middle
+	// Traverse till given position
+	for (int i = 0; temp != NULL && i < ind; i++) {
+		prev = temp;
+		temp = temp->next;
+	}
+
+	// If given index is found, delete node
+	if (temp != NULL && prev != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else {
+		return E_INVALID_INDEX;
+	}
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Removes the Node with the specified data value from the list.
+/// </summary>
+/// <param name="list"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Remove (struct LinkedList* list, int data) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	// Store head node
+	struct Node* temp = list->head;
+	struct Node* prev = NULL;
+
+	if (temp == NULL)
+		return E_EMPTY_LIST;
+
+	// If head node itself holds the key to be deleted
+	if (temp != NULL && temp->data == data) {
+
+		// Changed head
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+
+	// Search for the key to be deleted, keep track of the
+	// previous node as we need to change 'prev->next'
+	while (temp != NULL && temp->data != data) {
+		prev = temp;
+		temp = temp->next;
+	}
+
+	// If key was not present in linked list
+	if (temp != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else
+		return E_INVALID_ELEMENT;
+
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Counts the number of nodes in the list.
+/// </summary>
+/// <param name="list"></param>
+/// <returns>count/Error code (if any)</returns>
+int Count (struct LinkedList* list) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	int count = 0;
+	while (list->head != NULL) {
+		count++;
+		list->head = (list->head)->next;
+	}
+	return count;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Returns the value at a specified index if such index exists.
+/// </summary>
+/// <param name="list"></param>
+/// <param name="ind"></param>
+/// <returns>data</returns>
+int Get (struct LinkedList* list, int ind, int* data) {
+
+	if (list->isDeleted == true) {
+		*data = NULL;
+		return E_LIST_DNE;
+	}
+
+	struct Node* current = list->head;
+
+	if (current == NULL) {
+		*data = NULL;
+		return E_EMPTY_LIST;
+	}
+
+	// the index of the 
+	// node we're currently 
+	// looking at 
+	int count = 0;
+	while (current != NULL) {
+		if (count == ind) {
+			*data = current->data;
+			return 0;
+		}
+		count++;
+		current = current->next;
+	}
+
+	*data = NULL;
+	return E_INVALID_INDEX;

```

Comment:
You should check for invalid index before. In this case it would be more appropriate to call it Element for found error or something similar.